phys2bids/physio_obj.py

```diff
@@ -0,0 +1,321 @@
+    """
+    if len(var) > data_size:
+        var = var[:data_size]
+
+    if len(var) < data_size:
+        var = var + [token] * (data_size - len(var))
+
+    return var
+
+
+class blueprint_input():
+    """
+    Main input object for phys2bids.
+    Contains the blueprint to be populated.
+    !!! Pay attention: there's rules on how to populate this object.
+    See below ("Attention") !!!
+
+    Input (Properties)
+    ------------------
+    timeseries : (ch, [tps]) list
+        List of numpy 1d arrays - one for channel, plus one for time.
+        Time channel has to be the first, trigger the second.
+        Contains all the timeseries recorded.
+        Supports different frequencies!
+    freq : (ch) list of floats
+        List of floats - one per channel.
+        Contains all the frequencies of the recorded channel.
+        Support different frequencies!
+    ch_name : (ch) list of strings
+        List of names of the channels - can be the header of the columns
+        in the output files.
+    units : (ch) list of strings
+        List of the units of the channels.
+
+    Properties
+    ---------------------
+    ch_amount: int
+        Number of channels (ch).
+
+    Methods
+    -------
+    return_index:
+        Returns the proper list entry of all the
+        properties of the object, given an index.
+    delete_at_index:
+        Returns all the proper list entry of the
+        properties of the object, given an index.
+    check_trigger_amount:
+        Counts the amounts of triggers and corrects time offset
+        in "time" ndarray. Also adds property ch_amount.
+
+    Attention
+    ---------
+    The timeseries (and as a consequence, all the other properties)
+    should start with an entry for time and an entry for trigger.
+    Both should have the same length - hence same sampling. Meaning:
+    - timeseries[0] → ndarray representing time
+    - timeseries[1] → ndarray representing trigger
+    - timeseries[0].shape == timeseries[1].shape
+
+    As a consequence:
+    - freq[0] == freq[1]
+    - ch_name[0] = 'time'
+    - ch_name[1] = 'trigger'
+    - units[0] = 's'
+    - Actual number of channels (ANC) +1 <= ch_amount <= ANC +2
+    """
+    def __init__(self, timeseries, freq, ch_name, units):
+        self.timeseries = is_valid(timeseries, list, list_type=np.ndarray)
+        self.ch_amount = len(self.timeseries)
+        self.freq = has_size(is_valid(freq, list,
+                                      list_type=(int, float)),
+                             self.ch_amount, 0)
+        self.ch_name = has_size(ch_name, self.ch_amount, 'unknown')
+        self.units = has_size(units, self.ch_amount, '[]')
+
+    def rename_channels(cls, new_names, ch_trigger=None):
+        """
+        Renames the channels.
+
+        Input
+        -----
+        new_names: list of str
+            New names for channels.
+        ch_trigger:
+            Number of the channel containing the trigger.
```

Comment:
You're missing `cls`.

Reply:
Actually, if by `cls` you mean the object, the input should be `self`.

Reply:
Not according to pep8, at least [here](https://www.python.org/dev/peps/pep-0008/#id46)

Reply:
Should I specify "self" in the docstrings? I wouldn't like the user to get confused by it.

Reply:
Maybe @rmarkello will know better but the way I've always seen this is `self`.